### PR TITLE
idr-py: add recipe from PyPI

### PR DESCRIPTION
**Pull request by joshmoore**
This library is used for accessing https://idr.openmicroscopy.org, will be used in concert with an environment file: https://github.com/IDR/idr-py/pull/11, and depends on `python-omero` which we [maintain](https://github.com/bioconda/bioconda-recipes/pull/8187).

 * Created via `conda skeleton pypi idr-py`
 * Dependency on python-omero added manually

----

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/idr-py/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+source:
+  fn: '{{ name }}-{{ version }}.{{ file_ext }}'
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+  skip: true  # [py3k]
```
Comment (nsoranzo) [inline recipes/idr-py/meta.yaml]: [Guidelines](https://bioconda.github.io/guidelines.html#python) recommend:
```yaml
  skip: True  # [not py27]
```

File: recipes/idr-py/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+requirements:
+  host:
+    - python
+    - setuptools
+  run:
+    - python >=2.7,<3
```
Comment (nsoranzo) [inline recipes/idr-py/meta.yaml]: Just `- python` here should be fine.

File: recipes/idr-py/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ file_ext }}
+  '{{ hash_type }}': '{{ hash_value }}'
+
+build:
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record=record.txt
+  skip: True  # [not py27]
+
+requirements:
+  host:
```
Comment (nsoranzo) [inline recipes/idr-py/meta.yaml]: This should be `build:`, no?

File: recipes/idr-py/meta.yaml (hunk 1 above)
Comment (joshmoore) [reply recipes/idr-py/meta.yaml]: Thanks, @nsoranzo. Unfortunately seeing the same error with this change:

```
/opt/conda/lib/python3.6/site-packages/idr/tests/test_attributes.py
```

(I had tried various variants of this and force pushed them away)

File: recipes/idr-py/meta.yaml (hunk 3 above)
Comment (joshmoore) [reply recipes/idr-py/meta.yaml]: Interesting. This seems to have come from the initial code generation. I'll certainly give `build:` a try.

Comment (nsoranzo) [reply recipes/idr-py/meta.yaml]: It builds successfully now, at least. Some test fails, but I think you can work from there.

**Comment by joshmoore** [comment]
@nsoranzo : `s/host/build/` did the trick. With a subsequent matplotlib issue handled, CI is now green. Thanks for the help.

@bioconda/core : any comments on a new recipe?

**Comment by daler** [comment]
Looks good; I would say the only issue is a messy description field. We don't currently render this anywhere, so it's really just a style choice. Up to you -- merge when ready, and thanks!

**Comment by joshmoore** [comment]
We should have fairly regular updates; I'll incrementally improve the description (and perhaps open a PR for `conda skeleton...`) Thanks all.

**Pull request merged**